powerscale/provider/synciq_global_settings_resource.go

```diff
@@ -0,0 +1,578 @@
+					},
+					"pool": schema.StringAttribute{
+						Optional:            true,
+						Computed:            true,
+						Description:         "The pool to restrict replication policies to.",
+						MarkdownDescription: "The pool to restrict replication policies to.",
+					},
+				},
+			},
+			"tw_chkpt_interval": schema.Int64Attribute{
+				Optional:            true,
+				Computed:            true,
+				Description:         "The interval (in seconds) in which treewalk syncs are forced to checkpoint.",
+				MarkdownDescription: "The interval (in seconds) in which treewalk syncs are forced to checkpoint.",
+			},
+			"password": schema.StringAttribute{
+				Optional:            true,
+				Description:         "The password for cluster authentication",
+				MarkdownDescription: "The password for cluster authentication",
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(1, 255),
+				},
+			},
+		},
+	}
+}
+
+// Configure configures the resource.
+func (r *SyncIQGlobalSettingsResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	pscaleClient, ok := req.ProviderData.(*client.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = pscaleClient
+}
+
+// Create allocates the resource.
+func (r *SyncIQGlobalSettingsResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var diags diag.Diagnostics
+	tflog.Info(ctx, "Creating SyncIQ Global Settings resource state")
+	// Read Terraform plan into the model
+	var plan models.SyncIQGlobalSettingsModel
+	var toUpdate powerscale.V16SyncSettingsExtended
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	err := helper.ReadFromState(ctx, &plan, &toUpdate)
+	if err != nil {
+		errStr := constants.UpdateSyncIQGlobalSettingsErrorMsg + "with error: "
+		message := helper.GetErrorString(err, errStr)
+		resp.Diagnostics.AddError(
+			"Error updating synciq global settings",
+			fmt.Sprintf("Could not read synciq global setting param with error: %s", message),
+		)
+		return
+	}
+
+	if !plan.ReportEmail.IsUnknown() {
+		diags.Append(plan.ReportEmail.ElementsAs(ctx, &toUpdate.ReportEmail, true)...)
+	}
+
+	if toUpdate.SourceNetwork != nil {
+		if (toUpdate.SourceNetwork.Subnet == "" && toUpdate.SourceNetwork.Pool != "") || (toUpdate.SourceNetwork.Subnet != "" && toUpdate.SourceNetwork.Pool == "") {
+			resp.Diagnostics.AddError(
+				"Valid value for both subnet and pool needs to be provided",
+				"Please either provide value for both subnet and pool or else keep both as empty",
```

Comment:
Looks like Create and Update logic same so can we write one common function and same for Read import if possible

Reply:
Done